mne/preprocessing/ica.py

```diff
@@ -1486,21 +1488,33 @@ def _pick_sources(self, data, include, exclude):
         logger.info('Zeroing out %i ICA components'
                     % (n_components - len(sel_keep)))
 
-        unmixing = np.eye(_n_pca_comp)
-        unmixing[:n_components, :n_components] = self.unmixing_matrix_
-        unmixing = np.dot(unmixing, self.pca_components_[:_n_pca_comp])
+        if (self.method == 'imported_eeglab' and
+           np.diff(self.unmixing_matrix_.shape)[0]):
+            # When \PCA reduction is used in EEGLAB, runica returns
+            # weights= weights*sphere*eigenvectors(:,1:ncomps)';
+            # sphere = eye(urchans);
+            # such that we cannot separate the unmixing matrix and
+            # the PCA components. For that reason, the composite
+            # unmixing matrix takes a rectangular shape.
+            unmixing = self.unmixing_matrix_
+            mixing = self.mixing_matrix_
+
+        else:
+            unmixing = np.eye(_n_pca_comp)
+            unmixing[:n_components, :n_components] = self.unmixing_matrix_
+            unmixing = unmixing @ self.pca_components_[:_n_pca_comp]
 
-        mixing = np.eye(_n_pca_comp)
-        mixing[:n_components, :n_components] = self.mixing_matrix_
-        mixing = np.dot(self.pca_components_[:_n_pca_comp].T, mixing)
+            mixing = np.eye(_n_pca_comp)
+            mixing[:n_components, :n_components] = self.mixing_matrix_
+            mixing = self.pca_components_[:_n_pca_comp].T @ mixing
 
         if _n_pca_comp > n_components:
             sel_keep = np.concatenate(
                 (sel_keep, range(n_components, _n_pca_comp)))
 
-        proj_mat = np.dot(mixing[:, sel_keep], unmixing[sel_keep, :])
+        proj_mat = mixing[:, sel_keep] @ unmixing[sel_keep, :]
 
-        data = np.dot(proj_mat, data)
+        data = proj_mat @ data
```

Comment:
Love all of the `@`s

Reply:
Yeah.... I read that it was now the recommended way. And it is definitly more readable. So I though why not update this part of code on the way... :)